cob/cob.c

```diff
@@ -493,6 +499,74 @@ M0_INTERNAL void m0_cob_bc_iterator_fini(struct m0_cob_bc_iterator *it)
 	m0_free(it->ci_rec);
 }
 
+M0_INTERNAL int m0_cob_bc_entries_dump(struct m0_cob_domain *cdom,
+				       struct m0_buf        *out_keys,
+				       struct m0_buf        *out_recs,
+				       uint32_t             *out_count)
+{
+	struct m0_cob_bc_iterator it;
+	struct m0_list            entry_list;
+	struct m0_list_link      *list_pos;
+	int                       rc;
+	void                     *key_cursor;
+	void                     *rec_cursor;
+
+	M0_PRE(out_keys == NULL);
+	M0_PRE(out_recs == NULL);
+
+	if (cdom->cd_bytecount.bb_root != NULL)
+		return M0_ERR(-ENOENT);
+
+	*out_count = 0;
+	m0_be_btree_cursor_init(&it.ci_cursor, &cdom->cd_bytecount);
+	rc = m0_be_btree_cursor_first_sync(&it.ci_cursor);
+
+	while (rc != -ENOENT) {
+		struct bc_entry *entry;
+
+		M0_ALLOC_PTR(entry);
+		m0_list_init(&entry_list);
```

Comment:
Add a todo to use tlist here.

Reply:
Done in https://github.com/Seagate/cortx-motr/pull/1499